vyper/builtins/functions.py

```diff
@@ -2363,7 +2363,16 @@ def infer_kwarg_types(self, node):
         for kwarg in node.keywords:
             kwarg_name = kwarg.arg
             validate_expected_type(kwarg.value, self._kwargs[kwarg_name].typ)
-            ret[kwarg_name] = get_exact_type_from_node(kwarg.value)
+            if kwarg_name == "method_id":
```

Comment:
maybe out-of-scope but it would be much better for `validate_expected_type` to return the type it inferred

Reply:
I think I/we have tried this a few times before :smile: